### DISCUSSION TITLE
config/basic/

### DISCUSSION THREAD

**Discussion by giscus** [pengzhanbo/vuepress-theme-plume#96] [docs-comment]
# config/basic/

基础配置 plugins 类型：PlumeThemePluginOptions 默认值： {} 详情： 对主题内部使用的插件进行自定义配置。 主题使用的插件默认已进行了配置，大多数情况下您不需要进行任何修改，如果需要使用到细致的定制化，请查阅 hostname 类型： string 默认值： '' 详情： 部署站点域名。 当 hostname 配置为有...

https://theme-plume.vuejs.press/config/basic/

**Comment by Nick110** [comment]
有办法配置网站的favicon吗？好像没找到

**Reply by pengzhanbo** [reply]
参考下这个 <https://github.com/pengzhanbo/vuepress-theme-plume/blob/main/docs/.vuepress/config.ts#L16-L18>

**Reply by resetsix** [reply]
> 参考下这个 https://github.com/pengzhanbo/vuepress-theme-plume/blob/main/docs/.vuepress/config.ts#L16-L18

你好我想请教一下，我是在 head 里使用 png 图片配置网站 favicon 图标的并且能够正常显示（我不确定正确配置 favicon 的地方），但是默认会请求`http://localhost:8080/favicon.ico`并且是 404，所以我想请问一下应该在哪里配置 favicon或解决这个 404 问题，在 public 直接子目录放 favicon 文件吗

```ts
export default defineUserConfig({
  base: '/',
  lang: 'zh-CN',
  locales: xxx

  head: [
    // 配置站点图标
    [
      'link',
      {
        rel: 'icon',
        type: 'image/png',
        href: '/images/head/avatar.png', 
      },
    ],
  ],

});
```

**Reply by pengzhanbo** [reply]
站点图标，从兼容角度考虑，是需要在 public 下放一个 favicon.ico 文件的。head 中的 link icon 配置可以给当前页面，在受支持的浏览器中，提供更高清的图标显示。

**Reply by resetsix** [reply]
@pengzhanbo 明白了感谢。但我发现一个 BUG，不确定是 plume 还是 netlify 的原因。

经过测试，jpg、png、webp 三种格式的图片部署到netlify后，其中webp格式的图片在渲染的时候整张图片是花屏的，而jpg和png是正常。我在控制台检查图片发现返回的就是花屏图片，也没搜到相关 issure。我用其他博客部署 webp 图片到 netlify 又不会出现这种情况。

所以希望能进行排查一下。

**Comment by Nick110** [comment]
首页支持配置展示最近文章列表吗？

**Reply by pengzhanbo** [reply]
当前不支持，但是可以 参考 [自定义首页](https://theme-plume.vuejs.press/guide/custom-home/#%E8%87%AA%E5%AE%9A%E4%B9%89%E5%8C%BA%E5%9F%9F%E7%B1%BB%E5%9E%8B) ，以及 使用 `import { useLocalePostList } from 'vuepress-theme-plume/composables'` 获取文章列表，即可自行扩展首页内容。

**Reply by Nick110** [reply]
好的，多谢

**Comment by HyperCherry** [comment]
有方法直接禁用`blog`吗？我想仅搭建一个文档站点
```ts
export default defineThemeConfig({
  blog: false,
});
```
这种方式似乎无法使用，并且[#59](https://github.com/pengzhanbo/vuepress-theme-plume/issues/59#issuecomment-2012937646)中所提到的方式仍然会将`markdown`文档识别为`article`

**Reply by pengzhanbo** [reply]
这可能是在匹配规则上出现了冲突导致的，我会进行验证和修复。
当前 `blog` 是不支持 `false` 选项的，后续版本会考虑添加该支持。
如果希望仅提供文档支持，最简单的方式是，将[autoFrontmatter.permalink](https://theme-plume.vuejs.press/config/basic/#autofrontmatter) 设置为 false，主题将不再生成永久链接，而是恢复 VuePress 默认的以 文件夹路径结构作为页面访问地址。忽略 `blog`/`notes` 两个配置项，改用 `sidebar` 选项添加 侧边栏即可。
（`sidebar` 选项说明暂还未添加到文档中，但它目前是支持的，与 VuePress 默认主题的配置方式类似。）

**Comment by tianhairui** [comment]
Locale 配置中的 profile，示例代码 `theme: themePlume`，是否应该为 `theme: plumeTheme` 呢？它和指南中指定的名称不一致，且经过测试，有错误，可以抽空帮忙确认下。

**Reply by pengzhanbo** [reply]
感谢指正！是我在 profile 这部分的文档中写错了，应该是 `theme: plumeTheme`，我稍后修复下该错误。

**Comment by liuzhemax** [comment]
我想请教下这个svg图标怎么自定义呢。
我在plume.config.ts里面配置了：
  social: [
    { icon: 'github', link: '/' },
    { icon: {svg:'<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 32 32"><path fill="currentColor" d="M14 5C8.578 5 4 8.71 4 13.5c0 1.617.594 3.102 1.5 4.375l-.375 2.906l-.219 1.75l1.594-.687L9 20.812a11.4 11.4 0 0 0 4.156 1.125C13.886 24.892 16.976 27 20.5 27c1.453 0 2.781-.41 3.938-1.031l1.187.687l1.719.938l-.25-1.938l-.219-1.812C27.539 22.89 28 21.8 28 20.594c0-2.559-1.707-4.703-4.125-5.75c.086-.438.125-.875.125-1.344C24 8.71 19.422 5 14 5m0 2c4.52 0 8 3.008 8 6.5q0 .4-.063.781a8.3 8.3 0 0 0-1.437-.125c-3.8 0-7.09 2.461-7.469 5.781c-1.32-.132-2.531-.511-3.562-1.093l-.438-.219l-.437.188l-1.281.53l.218-1.53l.032-.438l-.25-.313C6.473 16.023 6 14.806 6 13.5C6 10.008 9.48 7 14 7m-3.438 2.156c-.777 0-1.406.63-1.406 1.406c0 .778.63 1.438 1.406 1.438c.778 0 1.407-.66 1.407-1.438s-.63-1.406-1.406-1.406zm6.688 0c-.777 0-1.438.63-1.438 1.406c0 .778.66 1.438 1.438 1.438s1.406-.66 1.406-1.438s-.629-1.406-1.406-1.406m3.25 7c3.14 0 5.5 2.094 5.5 4.438c0 .886-.32 1.691-.906 2.406l-.282.344l.063.406l.031.188l-.468-.25l-.5.343c-.934.606-2.125.969-3.438.969c-3.14 0-5.5-2.063-5.5-4.406c0-2.344 2.36-4.438 5.5-4.438M18 17.438c-.59 0-1.094.472-1.094 1.062s.504 1.063 1.094 1.063c.586 0 1.063-.473 1.063-1.063S18.59 17.437 18 17.437zm5.031 0a1.06 1.06 0 0 0-1.062 1.062c0 .59.472 1.063 1.062 1.063s1.063-.473 1.063-1.063s-.473-1.063-1.063-1.063z"/></svg>'}, link: '/' },
  ],
页面这个图标没显示出来呢，是什么原因呢？，应该怎么定义svg呢

**Reply by liuzhemax** [reply]
svg的值是什么呢，可以举个例子吗

**Reply by pengzhanbo** [reply]
`svg` 的值就是 SVG 图标的源码。
```
{ svg: '<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24"><path fill="currentColor" d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z"/></svg>' }
```

**Reply by liuzhemax** [reply]
我这样配置过，在plume.config.ts里面，页面还是没有显示呢  
social: [
    { icon: 'github', link: 'https://github.com/liuzhemax/' },
    { icon: { svg: '省略.....' }, link: '/about/' },
  ],
还有其他的配置影响这块吗

**Reply by liuzhemax** [reply]
navbarSocialInclude: ['github','qq',], 
这个配置需要额外对自定义的svg做配置吗，这是我目前的配置

**Reply by pengzhanbo** [reply]
social 主要是显示在 Blog 页面，以及支持一部分显示在 导航栏。但是配置了 `navbarSocialInclude` 之后怎么让 自定义的 social 显示在 导航栏，我还没有考虑过 🤔

**Reply by liuzhemax** [reply]
好的，谢谢

**Reply by pengzhanbo** [reply]
晚点我思考下这块内容怎么改，建议可以提一个 feature request 给我，国庆期间我应该没太多时间写代码。

**Reply by liuzhemax** [reply]
好的，感谢，不着急。

**Reply by pengzhanbo** [reply]
新版恩已添加支持，在自定义图标中，额外传入 `name`， `{svg: '', name: 'xxx' }`， 然后将 `name` 值传给 `navbarSocialInclude`
[查看文档](https://theme-plume.vuejs.press/config/basic/#social)

**Reply by liuzhemax** [reply]
好的，我看看，非常感谢！

**Reply by liuzhemax** [reply]
你好，测试出个小bug,在about页我写两个标题，test1和test2,跳转的时候test1跳转是正常，test2跳转会跳转到test1,并没有跳转到test2。
以下是我测试的示例：

  navbarSocialInclude: ['github','test1','test2',],
  social: [
    { icon: 'github', link: 'https://xxxx' },
     {
        icon: { svg: 'xxx', name: 'test1' },
        link: '/about/#test1'
      },
      {
        icon: { svg: 'xxx', name: 'test2' },
        link: '/about/#test2'
      },
  ],

**Comment by navms** [comment]
一个 bug：
    blog: {
      postList: false,
      categories: false
    },
禁用 categories 之后，还是会展示分类的图标

**Reply by navms** [reply]
另外就是禁用分类之后，我希望文章不要展示分类的信息，不知道是否可以做到

**Comment by TenOne506** [comment]
我的profile配置采用上面的配置	profile: {
      avatar: '/avatar.jpg',
      name: '张三',
      description: '此处无银三百两，隔壁王二不曾偷',
      circle: true,
      location: '杭州，中国',
      organization: 'xxx公司',
      layout: 'right',
    }, 其中name 和description 并没有做出任何更改。其他地方倒是进行了更改，我想知道还需要改那些

**Reply by pengzhanbo** [reply]
你是否在 locales 中有配置 profile ，以及在 locales 的同级也有 profile ？

**Reply by TenOne506** [reply]
我想我知道了问题在哪了，我在参考指南中侧边栏博客那一章的环节中，将profile写入了.vuepress/config.ts。实际上在项目生成的时候，profile在.vuepress/plume.config中已经自动生成了。应该优先修改.vuepress/plume.config中的profile,或者两个保留一个
